### PR TITLE
add content about nt-dml for insert and update

### PR DESCRIPTION
**Pull request by ekexium**
Signed-off-by: ekexium <eke@fastmail.com>

### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/issues/33485

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: non-transactional-dml.md [hunk 1]
```diff
@@ -9,11 +9,15 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
...
-非事务 DML 语句包括 `INSERT`、`UPDATE` 和 `DELETE`，TiDB 目前只支持非事务 `DELETE` 语句，详细的语法介绍见 [`BATCH`](/sql-statements/sql-statement-batch.md)。
+通常对于消耗内存过多的大事务，需要在应用中拆分 SQL 语句以绕过事务大小限制。非事务 DML 语句将这一功能集成到 TiDB 内核中，实现等价的效果。非事务 DML 语句的效果可以通过拆分 SQL 语句的结果来理解，`DRY RUN` 语法提供了预览拆分后语句的功能。
+
+非事务 DML 语句包括 `INSERT`、`REPLACE`、`UPDATE` 和 `DELETE`，详细的语法介绍见 [`BATCH`](/sql-statements/sql-statement-batch.md)。
 
 > **注意：**
 >
 > 非事务 DML 语句不保证该语句的原子性和隔离性，不能认为它和原始 DML 语句等价。
+> 尤其注意不应当对任意的 DML 语句，改写成非事务 DML 后假设其行为与原来一致。
+> 使用非事务 DML 前需要分析其拆分后的语句是否会互相影响。
```
Comment (Oreoxmt) [inline non-transactional-dml.md:20]:
```suggestion
> - 非事务 DML 语句不保证该语句的原子性和隔离性，不能认为它和原始 DML 语句等价。
> - 在任意 DML 语句改写为非事务 DML 语句后，不应假设其行为与原来一致。
> - 使用非事务 DML 前需要分析其拆分后的语句是否会互相影响。
```

File: non-transactional-dml.md [hunk 2]
```diff
@@ -31,11 +35,24 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
...
+
+```sql
+create table t(id int, v int, key(id))
+create table t2(id int, v int, key(id))
+insert into t values (1, 1), (2, 2), (3, 3)
+insert into t2 values (1, 1), (2, 2), (4, 4)
+batch on test.t.id limit 1 update t join t2 on t.id=t2.id set t2.id = t2.id+1
+select * from t2 -- (4, 1) (4, 2) (4, 4)
+```
+
```
Comment (Oreoxmt) [inline non-transactional-dml.md:55]:
```suggestion
```

File: non-transactional-dml.md [hunk 3]
```diff
@@ -104,6 +121,44 @@ SELECT * FROM t;
...
+```sql
+INSERT INTO t2 VALUES (1,1),(3,3),(5,5);
+```
+
+```sql
+Query OK, 3 rows affected
+```
+
+然后进行涉及多表 join 的更新，注意拆分列的指定需要完整的数据库名、表名、列名。
```
Comment (Oreoxmt) [inline non-transactional-dml.md:142]:
```suggestion
然后进行涉及多表 join 的更新（表 `t1` 和 `t2`）。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名 （`test.t.id`)。
```

File: non-transactional-dml.md [hunk 4]
```diff
@@ -104,6 +121,44 @@ SELECT * FROM t;
...
+CREATE TABLE t2(id int, v int, key(id));
+```
+
+```sql
+Query OK, 0 rows affected
+```
+
+```sql
+INSERT INTO t2 VALUES (1,1),(3,3),(5,5);
+```
```
Comment (Oreoxmt) [inline non-transactional-dml.md:136]:
````suggestion
```sql
CREATE TABLE t2(id int, v int, key(id));
INSERT INTO t2 VALUES (1,1),(3,3),(5,5);
```
````

File: non-transactional-dml.md [hunk 5]
```diff
@@ -104,6 +121,44 @@ SELECT * FROM t;
...
+
+```sql
+Query OK, 3 rows affected
+```
+
```
Comment (Oreoxmt) [inline non-transactional-dml.md:141]:
```suggestion
```

File: sql-statements/sql-statement-batch.md [hunk 6]
```diff
@@ -7,20 +7,44 @@ summary: TiDB 数据库中 BATCH 的使用概况。
...
 BATCH 语句将一个 DML 语句拆成多个语句在内部执行，因此**不保证**事务的原子性和隔离性，是一个“非事务”语句。
 
-目前 BATCH 语句仅支持 `DELETE`。
+目前 BATCH 语句支持 `INSERT`、`REPLACE`、`UPDATE`、`DELETE`。
 
 BATCH 语句在某一列将 DML 语句涉及的范围划分为多个区间，在每个区间执行一条 SQL。
 
 详细的说明和使用限制见[非事务语句](/non-transactional-dml.md)。
 
+在涉及多表 join 时，BATCH 语法中的拆分列要求用完整的路径写明以避免歧义。如
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-batch.md:16]:
```suggestion
在涉及多表 join 时，`BATCH` 语法中指定拆分列时需要指明完整的路径以避免歧义，如：
```

**Comment by Oreoxmt** [comment]
@you06 PTAL

File: sql-statements/sql-statement-batch.md [hunk 7]
```diff
@@ -7,20 +7,44 @@ summary: TiDB 数据库中 BATCH 的使用概况。
...
+
+```sql
+BATCH ON test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3.v FROM t2 JOIN t3 ON t2.id=t3.id"
```
Comment (you06) [inline sql-statements/sql-statement-batch.md:19]:
```suggestion
BATCH ON test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3.v FROM t2 JOIN t3 ON t2.id=t3.id;
```

File: sql-statements/sql-statement-batch.md [hunk 8]
```diff
@@ -7,20 +7,44 @@ summary: TiDB 数据库中 BATCH 的使用概况。
...
+```
+
+上面这条语句的拆分列用 `test.t2.id` 指明，不具有歧义。如果写成如下形式
+
+```sql
+batch on id limit 1 insert into t select t2.id, t2.v, t3.v from t2 join t3 on t2.id=t3.id"
```
Comment (you06) [inline sql-statements/sql-statement-batch.md:25]:
```suggestion
batch on id limit 1 insert into t select t2.id, t2.v, t3.v from t2 join t3 on t2.id=t3.id;
```

**Review by you06 — APPROVED** [review]

File: non-transactional-dml.md [hunk 9]
```diff
@@ -9,11 +9,15 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
 
 非事务 DML 语句是将一个普通 DML 语句拆成多个 SQL 语句（即多个 batch）执行，以牺牲事务的原子性和隔离性为代价，增强批量数据处理场景下的性能和易用性。
 
-非事务 DML 语句包括 `INSERT`、`UPDATE` 和 `DELETE`，TiDB 目前只支持非事务 `DELETE` 语句，详细的语法介绍见 [`BATCH`](/sql-statements/sql-statement-batch.md)。
+通常对于消耗内存过多的大事务，需要在应用中拆分 SQL 语句以绕过事务大小限制。非事务 DML 语句将这一功能集成到 TiDB 内核中，实现等价的效果。非事务 DML 语句的效果可以通过拆分 SQL 语句的结果来理解，`DRY RUN` 语法提供了预览拆分后语句的功能。
```
Comment (Oreoxmt) [inline non-transactional-dml.md:12]:
```suggestion
通常，消耗内存过多的大事务需要在应用中拆分 SQL 语句以绕过事务大小限制。非事务 DML 语句将这一功能集成到 TiDB 内核中，实现等价的效果。非事务 DML 语句的执行效果可以通过拆分 SQL 语句的结果来理解，`DRY RUN` 语法提供了预览拆分后语句的功能。
```

File: non-transactional-dml.md [hunk 10]
```diff
@@ -31,8 +35,20 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
...
+    - 拆分列也不应该用于 Join key。例如，下面示例将拆分列 `test.t.id` 作为 Join key，导致一个非事务 `UPDATE` 语句多次更新同一行：
+
+    ```sql
+    create table t(id int, v int, key(id));
+    create table t2(id int, v int, key(id));
+    insert into t values (1, 1), (2, 2), (3, 3);
+    insert into t2 values (1, 1), (2, 2), (4, 4);
+    batch on test.t.id limit 1 update t join t2 on t.id=t2.id set t2.id = t2.id+1;
+    select * from t2; -- (4, 1) (4, 2) (4, 4)
+    ```
```
Comment (Oreoxmt) [inline non-transactional-dml.md:50]:
````suggestion
    ```sql
    CREATE TABLE t(id int, v int, key(id));
    CREATE TABLE t2(id int, v int, key(id));
    INSERT INTO t VALUES (1, 1), (2, 2), (3, 3);
    INSERT INTO t2 VALUES (1, 1), (2, 2), (4, 4);
    BATCH ON test.t.id LIMIT 1 UPDATE t JOIN t2 ON t.id = t2.id SET t2.id = t2.id+1;
    SELECT * FROM t2; -- (4, 1) (4, 2) (4, 4)
    ```
````

File: non-transactional-dml.md [hunk 11]
```diff
@@ -104,6 +120,33 @@ SELECT * FROM t;
 1 row in set
 ```
 
+以下示例说明多表 join 的使用方法。首先创建表 `t2` 并插入数据。
+
+```sql
+CREATE TABLE t2(id int, v int, key(id));
+INSERT INTO t2 VALUES (1,1),(3,3),(5,5);
```
Comment (Oreoxmt) [inline non-transactional-dml.md:127]:
```suggestion
INSERT INTO t2 VALUES (1,1), (3,3), (5,5);
```

File: non-transactional-dml.md [hunk 12]
```diff
@@ -104,6 +120,33 @@ SELECT * FROM t;
...
+```
+
+然后进行涉及多表 join 的更新（表 `t1` 和 `t2`）。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名 （`test.t.id`)。
+
+```sql
+BATCH ON test.t.id LIMIT 1 UPDATE t JOIN t2 ON t.id=t2.id SET t2.id = t2.id+1
```
Comment (Oreoxmt) [inline non-transactional-dml.md:133]:
```suggestion
BATCH ON test.t.id LIMIT 1 UPDATE t JOIN t2 ON t.id = t2.id SET t2.id = t2.id+1;
```

File: sql-statements/sql-statement-batch.md [hunk 13]
```diff
@@ -7,20 +7,40 @@ summary: TiDB 数据库中 BATCH 的使用概况。
...
+目前 BATCH 语句支持 `INSERT`、`REPLACE`、`UPDATE`、`DELETE`。
 
 BATCH 语句在某一列将 DML 语句涉及的范围划分为多个区间，在每个区间执行一条 SQL。
 
 详细的说明和使用限制见[非事务语句](/non-transactional-dml.md)。
 
+在涉及多表 join 时，`BATCH` 语法中指定拆分列时需要指明完整的路径以避免歧义，如：
+
+```sql
+BATCH ON test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3.v FROM t2 JOIN t3 ON t2.id=t3.id;
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-batch.md:19]:
```suggestion
BATCH ON test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3.v FROM t2 JOIN t3 ON t2.id = t3.id;
```

File: sql-statements/sql-statement-batch.md [hunk 14]
```diff
@@ -7,20 +7,40 @@ summary: TiDB 数据库中 BATCH 的使用概况。
...
+```
+
+上面这条语句的拆分列用 `test.t2.id` 指明，不具有歧义。如果写成如下形式，则会报错：
+
+```sql
+BATCH ON id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3.v FROM t2 JOIN t3 ON t2.id=t3.id;
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-batch.md:25]:
```suggestion
BATCH ON id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3.v FROM t2 JOIN t3 ON t2.id = t3.id;
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

File: non-transactional-dml.md [hunk 15]
```diff
@@ -104,6 +120,33 @@ SELECT * FROM t;
...
 ```
 
+以下示例说明多表 join 的使用方法。首先创建表 `t2` 并插入数据。
+
+```sql
+CREATE TABLE t2(id int, v int, key(id));
+INSERT INTO t2 VALUES (1,1),(3,3),(5,5);
+```
+
+然后进行涉及多表 join 的更新（表 `t1` 和 `t2`）。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名 （`test.t.id`)。
```
Comment (Oreoxmt) [inline non-transactional-dml.md:130]:
```suggestion
然后进行涉及多表 join 的更新。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名 （`test.t.id`)。
```

File: non-transactional-dml.md [hunk 16]
```diff
@@ -104,6 +120,33 @@ SELECT * FROM t;
...
+
+```sql
+BATCH ON test.t.id LIMIT 1 UPDATE t JOIN t2 ON t.id=t2.id SET t2.id = t2.id+1
+```
+
+```sql
```
Comment (Oreoxmt) [inline non-transactional-dml.md:136]:
```suggestion
查看更新后表的数据：

```sql
```

File: sql-statements/sql-statement-batch.md [hunk 17]
```diff
@@ -7,20 +7,40 @@ summary: TiDB 数据库中 BATCH 的使用概况。
...
 
 详细的说明和使用限制见[非事务语句](/non-transactional-dml.md)。
 
+在涉及多表 join 时，`BATCH` 语法中指定拆分列时需要指明完整的路径以避免歧义，如：
+
+```sql
+BATCH ON test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3.v FROM t2 JOIN t3 ON t2.id=t3.id;
+```
+
+上面这条语句的拆分列用 `test.t2.id` 指明，不具有歧义。如果写成如下形式，则会报错：
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-batch.md:22]:
```suggestion
上面这条语句的拆分列用 `test.t2.id` 指明，不具有歧义。如果写成如下 `id` 的形式，则会报错：
```

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Review by TomShawn — APPROVED** [review]
LGTM

File: non-transactional-dml.md [hunk 18]
```diff
@@ -9,11 +9,15 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
 
 非事务 DML 语句是将一个普通 DML 语句拆成多个 SQL 语句（即多个 batch）执行，以牺牲事务的原子性和隔离性为代价，增强批量数据处理场景下的性能和易用性。
 
-非事务 DML 语句包括 `INSERT`、`UPDATE` 和 `DELETE`，TiDB 目前只支持非事务 `DELETE` 语句，详细的语法介绍见 [`BATCH`](/sql-statements/sql-statement-batch.md)。
+通常，消耗内存过多的大事务需要在应用中拆分 SQL 语句以绕过事务大小限制。非事务 DML 语句将这一功能集成到 TiDB 内核中，实现等价的效果。非事务 DML 语句的执行效果可以通过拆分 SQL 语句的结果来理解，`DRY RUN` 语法提供了预览拆分后语句的功能。
```
Comment (TomShawn) [inline non-transactional-dml.md:12]:
```suggestion
通常，对于消耗内存过多的大事务，你需要在应用中拆分 SQL 语句以绕过事务大小限制。非事务 DML 语句将这一功能集成到 TiDB 内核中，实现等价的效果。非事务 DML 语句的执行效果可以通过拆分 SQL 语句的结果来理解，`DRY RUN` 语法提供了预览拆分后语句的功能。
```

File: non-transactional-dml.md [hunk 19]
```diff
@@ -104,6 +120,35 @@ SELECT * FROM t;
...
 ```
 
+以下示例说明多表 join 的使用方法。首先创建表 `t2` 并插入数据。
+
+```sql
+CREATE TABLE t2(id int, v int, key(id));
+INSERT INTO t2 VALUES (1,1), (3,3), (5,5);
+```
+
+然后进行涉及多表 join 的更新（表 `t` 和 `t2`）。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名 （`test.t.id`)。
```
Comment (TomShawn) [inline non-transactional-dml.md:130]:
```suggestion
然后进行涉及多表 join 的更新（表 `t` 和 `t2`）。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名（`test.t.id`)。
```

**Review by TomShawn — APPROVED** [review]
LGTM

File: non-transactional-dml.md [hunk 20]
```diff
@@ -31,8 +35,20 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
 
 - 确保该语句不需要原子性，即允许执行结果中，一部分行被修改，而一部分行没有被修改。
 - 确保该语句具有幂等性，或是做好准备根据错误信息对部分数据重试。如果系统变量 `tidb_redact_log = 1` 且 `tidb_nontransactional_ignore_error = 1`，则该语句必须是幂等的。否则语句部分失败时，无法准确定位失败的部分。
-- 确保该语句将要操作的数据没有其它并发的写入，即不被其它语句同时更新。否则可能出现漏删、多删等非预期的现象。
+- 确保该语句将要操作的数据没有其它并发的写入，即不被其它语句同时更新。否则可能出现漏写、多写、重复修改同一行等非预期的现象。
 - 确保该语句不会修改语句自身会读取的内容，否则后续的 batch 读到之前 batch 写入的内容，容易引起非预期的情况。
+    - 拆分列不应该在语句中更新。例如，对于一条非事务 `UPDATE` 语句，拆分后的 SQL 依次执行，前一 batch 的修改提交后被后一 batch 读到，导致同一行数据被多次修改。
```
Comment (SunRunAway) [inline non-transactional-dml.md:40]:
```suggestion
    - 在使用非事务 `INSERT INTO SELECT` 时，尽量不要在插入时修改拆分列。
        - 不推荐： `BATCH ON test.t.id LIMIT 10000 INSERT INTO t SELECT id+1, value FROM t`
        - 推荐：`BATCH ON test.t.id LIMIT 10000 INSERT INTO t SELECT id, value FROM t`
        - 当 id 列具有 AUTO_INCREMENT 属性时，推荐：`BATCH ON test.t.id LIMIT 10000 INSERT INTO t(value) SELECT value FROM t`
    - 在使用非事务 `UPDATE`、`INSERT ... ON DUPLICATE KEY UPDATE`、`REPLACE INTO` 拆分列不应该在语句中更新。例如，对于一条非事务 `UPDATE` 语句，拆分后的 SQL 依次执行，前一 batch 的修改提交后被后一 batch 读到，导致同一行数据被多次修改。
        - 此语句不支持： `BATCH ON test.t.id LIMIT 10000 UPDATE t SET test.t.id = test.t.id-1`
        - 不推荐：`BATCH ON test.t.id LIMIT 1 INSERT INTO t SELECT id+1, value FROM t ON DUPLICATE KEY UPDATE id = id + 1;`
```

Comment (Oreoxmt) [reply non-transactional-dml.md:40]: @ekexium PTAL

Comment (ekexium) [reply non-transactional-dml.md:40]: 我觉得 INSERT 也可以提一下原因和后果，改了可能导致重复插入

**Comment by Oreoxmt** [comment]
@SunRunAway PTAL

File: non-transactional-dml.md [hunk 21]
```diff
@@ -31,8 +35,27 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
 
 - 确保该语句不需要原子性，即允许执行结果中，一部分行被修改，而一部分行没有被修改。
 - 确保该语句具有幂等性，或是做好准备根据错误信息对部分数据重试。如果系统变量 `tidb_redact_log = 1` 且 `tidb_nontransactional_ignore_error = 1`，则该语句必须是幂等的。否则语句部分失败时，无法准确定位失败的部分。
-- 确保该语句将要操作的数据没有其它并发的写入，即不被其它语句同时更新。否则可能出现漏删、多删等非预期的现象。
+- 确保该语句将要操作的数据没有其它并发的写入，即不被其它语句同时更新。否则可能出现漏写、多写、重复修改同一行等非预期的现象。
 - 确保该语句不会修改语句自身会读取的内容，否则后续的 batch 读到之前 batch 写入的内容，容易引起非预期的情况。
+    - 在使用非事务 `INSERT INTO ... SELECT` 同一张表时，尽量不要在插入时修改拆分列，否则可能因为多个批次读取到同一行，导致重复插入：
```
Comment (Oreoxmt) [inline non-transactional-dml.md:40]:
```suggestion
    - 在使用非事务 `INSERT INTO ... SELECT` 处理同一张表时，尽量不要在插入时修改拆分列，否则可能因为多个批次读取到同一行，导致重复插入：
```

File: non-transactional-dml.md [hunk 22]
```diff
@@ -104,6 +127,35 @@ SELECT * FROM t;
...
+
+```sql
+CREATE TABLE t2(id int, v int, key(id));
+INSERT INTO t2 VALUES (1,1), (3,3), (5,5);
+```
+
+然后进行涉及多表 join 的更新（表 `t` 和 `t2`）。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名（`test.t.id`)。
+
+```sql
+BATCH ON test.t.id LIMIT 1 UPDATE t JOIN t2 ON t.id = t2.id SET t2.id = t2.id+1;
```
Comment (SunRunAway) [inline non-transactional-dml.md:140]: 这个例子是不是和上面的限制冲突了？
上面说拆分列也不应该用于 Join key， 可是这个例子就是。

File: sql-statements/sql-statement-batch.md [hunk 23]
```diff
@@ -7,20 +7,40 @@ summary: TiDB 数据库中 BATCH 的使用概况。
...
+目前 BATCH 语句支持 `INSERT`、`REPLACE`、`UPDATE`、`DELETE`。
 
 BATCH 语句在某一列将 DML 语句涉及的范围划分为多个区间，在每个区间执行一条 SQL。
 
 详细的说明和使用限制见[非事务语句](/non-transactional-dml.md)。
 
+在涉及多表 join 时，`BATCH` 语法中指定拆分列时需要指明完整的路径以避免歧义，如：
+
+```sql
+BATCH ON test.t2.id LIMIT 1 INSERT INTO t SELECT t2.id, t2.v, t3.v FROM t2 JOIN t3 ON t2.id = t3.id;
```
Comment (SunRunAway) [inline sql-statements/sql-statement-batch.md:19]: 例子存在拆分列是join key

File: non-transactional-dml.md (hunk 22 above)
Comment (ekexium) [reply non-transactional-dml.md]: 这个检查我们不做，实际上是为了防止用户误用在文档里加的这一条限制。虽然这里这样用并没有问题，不过还是改掉吧（

**Review by SunRunAway — APPROVED** [review]

File: non-transactional-dml.md [hunk 24]
```diff
@@ -183,16 +235,16 @@ BATCH ON id LIMIT 2 DELETE /*+ USE_INDEX(t)*/ FROM t where v < 6;
 建议按照以下步骤执行非事务 DML 语句：
 
 1. 选择合适的[划分列](#参数说明)。建议使用整数或字符串类型。
-2. （可选）在非事务 DML 语句中添加 `DRY RUN QUERY`，手动执行查询，确认 DML 语句影响的数据范围是否大体正确。
-3. （可选）在非事务 DML 语句中添加 `DRY RUN`，手动执行查询，检查拆分后的语句和执行计划。需要关注索引选择效率。
+2. 在非事务 DML 语句中添加 `DRY RUN QUERY`，手动执行查询，确认 DML 语句影响的数据范围是否大体正确。
+3. 在非事务 DML 语句中添加 `DRY RUN`，手动执行查询，检查拆分后的语句和执行计划。需要关注一条拆分后的语句是否有可能读到之前的语句执行写入的结果，否则容易造成异常现象。需要关注索引选择效率。如果是由 TiDB 自动选择拆分列，可能选择了会被修改的拆分列，需要尤其关注。
```
Comment (Oreoxmt) [inline non-transactional-dml.md:239]:
```suggestion
3. 在非事务 DML 语句中添加 `DRY RUN`，手动执行查询，检查拆分后的语句和执行计划。需要关注：

    - 一条拆分后的语句是否有可能读到之前的语句执行写入的结果，否则容易造成异常现象。
    - 索引选择效率。
    - 由 TiDB 自动选择的拆分列是否可能会被修改。

```

File: non-transactional-dml.md [hunk 25]
```diff
@@ -104,6 +127,35 @@ SELECT * FROM t;
...
 ```
 
+以下示例说明多表 join 的使用方法。首先创建表 `t2` 并插入数据。
+
+```sql
+CREATE TABLE t2(id int, v int, key(id));
+INSERT INTO t2 VALUES (1,1), (3,3), (5,5);
+```
+
+然后进行涉及多表 join 的更新（表 `t` 和 `t2`）。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名（`test.t.id`)。
```
Comment (Oreoxmt) [inline non-transactional-dml.md:137]:
```suggestion
然后进行涉及多表 join 的更新（表 `t` 和 `t2`）。需要注意的是，指定拆分列时需要完整的数据库名、表名和列名（`test.t._tidb_rowid`)。
```

File: non-transactional-dml.md [hunk 26]
```diff
@@ -9,11 +9,15 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
 
 非事务 DML 语句是将一个普通 DML 语句拆成多个 SQL 语句（即多个 batch）执行，以牺牲事务的原子性和隔离性为代价，增强批量数据处理场景下的性能和易用性。
 
-非事务 DML 语句包括 `INSERT`、`UPDATE` 和 `DELETE`，TiDB 目前只支持非事务 `DELETE` 语句，详细的语法介绍见 [`BATCH`](/sql-statements/sql-statement-batch.md)。
+通常，对于消耗内存过多的大事务，你需要在应用中拆分 SQL 语句以绕过事务大小限制。非事务 DML 语句将这一过程集成到 TiDB 内核中，实现等价的效果。非事务 DML 语句的执行效果可以通过拆分 SQL 语句的结果来理解，`DRY RUN` 语法提供了预览拆分后语句的功能。
+
+非事务 DML 语句包括 `INSERT`、`REPLACE`、`UPDATE` 和 `DELETE`，详细的语法介绍见 [`BATCH`](/sql-statements/sql-statement-batch.md)。
```
Comment (SunRunAway) [inline non-transactional-dml.md:14]: 这里要不要明确说成：
- `INSERT INTO ... SELECT`
- `REPLACE INTO ... SELECT`
- `UPDATE`
- `DELETE`

Comment (Oreoxmt) [inline non-transactional-dml.md:14]:
```suggestion
非事务 DML 语句包括：

- `INSERT INTO ... SELECT`
- `REPLACE INTO .. SELECT`
- `UPDATE`
- `DELETE`

详细的语法介绍见 [`BATCH`](/sql-statements/sql-statement-batch.md)。
```
@SunRunAway @ekexium PTAL

Comment (SunRunAway) [reply non-transactional-dml.md:14]: @ekexium 你的意见呢？

Comment (ekexium) [reply non-transactional-dml.md:14]: 合理。

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: non-transactional-dml.md [hunk 27]
```diff
@@ -31,8 +42,27 @@ summary: 以事务的原子性和隔离性为代价，将 DML 语句拆成多个
 
 - 确保该语句不需要原子性，即允许执行结果中，一部分行被修改，而一部分行没有被修改。
 - 确保该语句具有幂等性，或是做好准备根据错误信息对部分数据重试。如果系统变量 `tidb_redact_log = 1` 且 `tidb_nontransactional_ignore_error = 1`，则该语句必须是幂等的。否则语句部分失败时，无法准确定位失败的部分。
-- 确保该语句将要操作的数据没有其它并发的写入，即不被其它语句同时更新。否则可能出现漏删、多删等非预期的现象。
+- 确保该语句将要操作的数据没有其它并发的写入，即不被其它语句同时更新。否则可能出现漏写、多写、重复修改同一行等非预期的现象。
 - 确保该语句不会修改语句自身会读取的内容，否则后续的 batch 读到之前 batch 写入的内容，容易引起非预期的情况。
+    - 在使用非事务 `INSERT INTO ... SELECT` 处理同一张表时，尽量不要在插入时修改拆分列，否则可能因为多个 batch 读取到同一行，导致重复插入：
+        - 不推荐使用 `BATCH ON test.t.id LIMIT 10000 INSERT INTO t SELECT id+1, value FROM t;`
+        - 推荐使用 `BATCH ON test.t.id LIMIT 10000 INSERT INTO t SELECT id, value FROM t;`
```
Comment (TomShawn) [inline non-transactional-dml.md:49]: 可以合并到同一行吗？

**Review by TomShawn — APPROVED** [review]
rest LGTM

File: non-transactional-dml.md (hunk 27 above)
Comment (Oreoxmt) [reply non-transactional-dml.md:49]: image

从预览效果来看，分开两行可以比较清楚地看到推荐和不推荐的差异是什么

**Pull request merged**